src/twisted/python/test/test_url.py

```diff
@@ -5,12 +5,18 @@
 """
 Tests for L{twisted.python.url}.
 """
-
 from __future__ import unicode_literals
 
-from ..url import URL
+import warnings
+
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore", category=DeprecationWarning)
+    from ..url import URL
+
 unicode = type(u'')
 from unittest import TestCase
+from twisted import trial
```

Comment:
Actually just `from twisted.trial.unittest import SynchronousTestCase`, as glyph points out, since we don't need any `Deferred` handling here.